src/runtime/terminal.go

```diff
@@ -697,9 +672,33 @@ func (term *Terminal) TemplateCache() *cache.Template {
 	if shlvl, err := strconv.Atoi(val); err == nil {
 		tmplCache.SHLVL = shlvl
 	}
+}
+
+func (term *Terminal) loadTemplateCache() {
+	defer term.Trace(time.Now())
+
+	val, OK := term.sessionCache.Get(cache.TEMPLATECACHE)
+	if !OK {
+		return
+	}
+
+	tmplCache := term.tmplCache
+
+	err := json.Unmarshal([]byte(val), &tmplCache)
+	if err != nil {
+		term.Error(err)
+		return
+	}
+
+	tmplCache.Segments = tmplCache.SegmentsCache.ToConcurrent()
+}
+
+func (term *Terminal) Logs() string {
```

Comment:
where is this used?

Reply:
It's used in debugging. Actually, this is part of the existing code and I didn't touch it. Something raised by the Git diff algorithm. :)